api/typeMapper.go

```diff
@@ -0,0 +1,162 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// mapStructToType converts the given map struct into the desired target type
+// The target type has to be annotated with 'mapstructure' tags
+// Example:
+//
+// type MyTargetType struct {
+//  Field1 int `mapstructure:"field1"`
+//  Field2 string `mapstructure:"another_field"`
+// }
+func mapStructToType(source map[string]interface{}, target interface{}) error {
+
+	config := &mapstructure.DecoderConfig{
+		Result:           target,
+		WeaklyTypedInput: true,
+		ErrorUnused:      true,
+	}
+
+	decoder, err := mapstructure.NewDecoder(config)
+	if err != nil {
+		return err
+	}
+
+	if err := decoder.Decode(source); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// toTypeArray converts a given byte slice into the provided slice of one type
+// Example
+//
+// var vertices []Vertex
+//
+// err := toTypeArray(data, &vertices)
+func toTypeArray(input []byte, target interface{}) error {
+	if input == nil {
+		return fmt.Errorf("Data is nil")
+	}
+
+	parsedInput := make([]interface{}, 0)
+	if err := json.Unmarshal(input, &parsedInput); err != nil {
+		return err
+	}
+
+	// just for the primitive type
+	switch targetValue := target.(type) {
+	case *[]TypedValue:
+		values, err := toValues(parsedInput)
+		if err != nil {
+			return err
+		}
+		*targetValue = append(*targetValue, values...)
+		return nil
+	}
+
+	// handling of complex types
+	for _, element := range parsedInput {
+		mapStrct, ok := element.(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("Failed to cast %v (%T) into map[string]interface{}", element, element)
+		}
+
+		switch targetValue := target.(type) {
+		case *[]Property:
+			var property Property
```

Comment:
sadly i have no idea to avoid this.
i tried to use the type information to create a variable of that particular type using reflection but it was not successful

Reply:
As discussed this is fine. It seems to be more difficult to avoid this in Golang.